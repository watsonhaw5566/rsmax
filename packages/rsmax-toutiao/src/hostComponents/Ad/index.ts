import { createHostComponent } from '@rsmax/runtime';
import type React from 'react';
import type { BaseProps, GenericEvent } from '../../types/component';

export interface AdProps extends BaseProps {
  // 广告位 id，必填。需要预先在「小程序管理后台 > 流量主模块」配置生成。
  unitId: string;
  // 广告加载成功的回调方法 1.19.0
  onLoad?: (event: GenericEvent) => any;
  // 广告加载失败的回调方法 1.26.2
  onError?: (event: GenericEvent) => any;
  // 广告关闭的回调方法 1.31.0
  onClose?: (event: GenericEvent) => any;
  fixed?: boolean;
  type?: 'video' | 'large' | 'lImg' | 'rImg';
  scale?: number;
}

export const Ad: React.ComponentType<AdProps> = createHostComponent<AdProps>('ad');

Ad.defaultProps = {
  scale: 100,
};
