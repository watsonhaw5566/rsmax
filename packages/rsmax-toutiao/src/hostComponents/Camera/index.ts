import { createHostComponent } from '@rsmax/runtime';
import type React from 'react';

import type { BaseProps } from '../../types/component';

/**
 * camera
 * https://developer.open-douyin.com/docs/resource/zh-CN/mini-app/develop/component/media-component/camera
 */
export interface CameraProps extends BaseProps {
  mode?: 'normal' | 'scanCode';
  resolution?: string;
  devicePosition?: 'front' | 'back';
  flash?: 'on' | 'off';
  frameSize?: 'small' | 'medium' | 'large';
  onInitdone?: (e: any) => void;
  onError?: (e: any) => void;
  onStop?: (e: any) => void;
  onScancode?: (e: any) => void;
  onClick?: (e: any) => void;
}

export const Camera: React.ComponentType<CameraProps> = createHostComponent<CameraProps>('camera');
