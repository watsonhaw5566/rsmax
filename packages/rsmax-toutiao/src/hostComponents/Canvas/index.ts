import { createHostComponent } from '@rsmax/runtime';
import type React from 'react';

import type { BaseProps } from '../../types/component';

export interface CanvasProps extends BaseProps {
  canvasId: string;
}

export const Canvas: React.ComponentType<CanvasProps> = createHostComponent<CanvasProps>('canvas');
