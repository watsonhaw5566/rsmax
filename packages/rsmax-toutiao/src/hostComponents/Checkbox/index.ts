import { createHostComponent } from '@rsmax/runtime';
import type React from 'react';

import type { BaseProps } from '../../types/component';

export interface CheckboxProps extends BaseProps {
  name?: string;
  value?: string;
  disabled?: boolean;
  checked?: boolean;
  color?: string;
}

export const Checkbox: React.ComponentType<CheckboxProps> = createHostComponent<CheckboxProps>('checkbox');

Checkbox.defaultProps = {
  disabled: false,
  checked: false,
};
