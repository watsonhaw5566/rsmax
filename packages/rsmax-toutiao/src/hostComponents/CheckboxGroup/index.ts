import React from 'react';
import { createHostComponent } from '@rsmax/runtime';

import { BaseProps } from '../../types/component';

export interface CheckboxGroupProps extends BaseProps {
  name?: string;
  onChange?: (e: any) => void;
}

export const CheckboxGroup: React.ComponentType<CheckboxGroupProps> =
  createHostComponent<CheckboxGroupProps>('checkbox-group');
