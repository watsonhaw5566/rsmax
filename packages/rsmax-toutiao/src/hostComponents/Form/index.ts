import { createHostComponent } from '@rsmax/runtime';
import type React from 'react';

import type { BaseProps } from '../../types/component';

export interface FormProps extends BaseProps {
  onSubmit?: (e: any) => void;
  onReset?: (e: any) => void;
  reportSubmit?: boolean;
  catchSubmit?: (e: any) => void;
  catchReset?: (e: any) => void;
}

export const Form: React.ComponentType<FormProps> = createHostComponent<FormProps>('form');

Form.defaultProps = {
  reportSubmit: false,
};
