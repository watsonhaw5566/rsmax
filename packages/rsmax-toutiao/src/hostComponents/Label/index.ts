import { createHostComponent } from '@rsmax/runtime';
import type React from 'react';

import type { BaseProps } from '../../types/component';

export interface LabelProps extends BaseProps {
  for?: string;
}

export const Label: React.ComponentType<LabelProps> = createHostComponent<LabelProps>('label');
