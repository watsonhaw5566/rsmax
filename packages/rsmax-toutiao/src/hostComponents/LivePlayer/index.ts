import { createHostComponent } from '@rsmax/runtime';
import type React from 'react';

import type { BaseProps } from '../../types/component';

type SignatureProps = {
  enable: boolean;
  content: string;
  position: string;
  color: string;
};

/**
 * live-player
 * https://developer.open-douyin.com/docs/resource/zh-CN/mini-app/develop/component/media-component/live-player
 */
export interface LivePlayerProps extends BaseProps {
  src: string;
  autoplay?: boolean;
  muted?: boolean;
  orientation?: 'vertical' | 'horizontal';
  objectFit?: 'contain' | 'fillCrop';
  signature?: SignatureProps;
  showCastingButton?: boolean;
  onStatechange?: (e: any) => void;
  onFullscreenchange?: (e: any) => void;
  onError?: (e: any) => void;
  onClick?: (e: any) => void;
  children?: React.ReactNode;
}

export const LivePlayer: React.ComponentType<LivePlayerProps> = createHostComponent<LivePlayerProps>('live-player');
