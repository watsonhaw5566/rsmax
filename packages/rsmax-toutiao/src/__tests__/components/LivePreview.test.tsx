import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import { LivePreview } from '../../hostComponents';

describe('LivePreview', () => {
  it('render correctly', () => {
    const testRenderer = TestRenderer.create(<LivePreview awemeId={'123'} />);

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
