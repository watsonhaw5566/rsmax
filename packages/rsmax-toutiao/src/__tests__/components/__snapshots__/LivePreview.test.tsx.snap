// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LivePreview > render correctly 1`] = `
<live-preview
  awemeId="123"
/>
`;
